haystack/components/toolinvoker.py

```diff
@@ -0,0 +1,57 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import json
+from typing import List
+
+from haystack import component
+from haystack.dataclasses import ChatMessage
+from haystack.dataclasses.tool import Tool
+
+_FUNCTION_NAME_FAILURE = (
+    "I'm sorry, I tried to run a function that did not exist. Would you like me to correct it and try again?"
+)
+_FUNCTION_RUN_FAILURE = "Seems there was an error while running the function: {error}"
+
+
+# JUST A SLIGHTLY MODIFIED VERSION OF OpenAIFunctionCaller
```

Comment:
I'm not particularly sure about the design of this component, I mostly recycled `OpenAIFunctionCaller`. In any case, it will be created in `haystack-experimental`.